frontend/src/lib/StatusUtils.ts

```diff
@@ -116,3 +129,76 @@ function wasNodeCached(node: NodeStatus): boolean {
     ? false
     : artifacts.some(artifact => artifact.s3 && !artifact.s3.key.includes(node.id));
 }
+
+// separate these helper function for paritial v2 api integration
+export function hasFinishedV2(state?: V2beta1RuntimeState): boolean {
+  switch (state) {
+    case V2beta1RuntimeState.SUCCEEDED: // Fall through
+    case V2beta1RuntimeState.SKIPPED: // Fall through
+    case V2beta1RuntimeState.FAILED: // Fall through
+    case V2beta1RuntimeState.CANCELED:
+      return true;
+    case V2beta1RuntimeState.PENDING: // Fall through
+    case V2beta1RuntimeState.RUNNING: // Fall through
+    case V2beta1RuntimeState.CANCELING: // Fall through
+    case V2beta1RuntimeState.RUNTIMESTATEUNSPECIFIED:
+      return false;
+    default:
+      return false;
```

Comment:
I think we should throw exception or have some sort of warnings instead, this is an unexpected situation.

Reply:
Good suggestion. Add a warn and an error to handle any unexpected state.